### PR TITLE
Addressing more feedback from Marc Gauthier.

### PR DESCRIPTION
**Pull request by timsifive**
See https://lists.riscv.org/g/tech-debug/message/756

### CODE REVIEW

File: xml/hwbp_registers.xml [hunk 1]
```diff
@@ -419,17 +432,17 @@
         </field>
 
         <field name="m" bits="9" access="R/W" reset="0">
-            When set, every instruction completed or exception taken in M-mode decrements \FcsrIcountCount
-            by 1.
+            When set, every instruction completed in or exception taken from
```
Comment (pdonahue-ventana) [inline xml/hwbp_registers.xml]:
```suggestion
            When set, every instruction completed in or trap taken from
```

File: xml/hwbp_registers.xml [hunk 2]
```diff
@@ -419,17 +432,17 @@
...
+            M-mode decrements \FcsrIcountCount by 1.
         </field>
         <field name="0" bits="8" access="R" reset="0" />
         <field name="s" bits="7" access="R/W" reset="0">
-            When set, every instruction completed or exception taken in S-mode decrements \FcsrIcountCount
-            by 1.
+            When set, every instruction completed in or exception taken from
```
Comment (pdonahue-ventana) [inline xml/hwbp_registers.xml]:
```suggestion
            When set, every instruction completed in or trap taken from
```

File: xml/hwbp_registers.xml [hunk 3]
```diff
@@ -419,17 +432,17 @@
...
+            S-mode decrements \FcsrIcountCount by 1.
         </field>
         <field name="u" bits="6" access="R/W" reset="0">
-            When set, every instruction completed or exception taken in U-mode decrements \FcsrIcountCount
-            by 1.
+            When set, every instruction completed in or exception taken from
```
Comment (pdonahue-ventana) [inline xml/hwbp_registers.xml]:
```suggestion
            When set, every instruction completed in or trap taken from
```

File: core_debug.tex [hunk 4]
```diff
@@ -78,8 +78,15 @@ \section{Wait for Interrupt Instruction}
...
 
-A debugger can cause a halted hart to execute a single instruction and then
-re-enter Debug Mode by setting \FcsrDcsrStep before setting \FdmDmcontrolResumereq.
+\subsection{Step Bit In Dcsr}
+
+This method is only available to external debuggers, and is the preferred way
+to single step.
+
+An external debugger can cause a halted hart to execute a single instruction or
+exception
```
Comment (pdonahue-ventana) [inline core_debug.tex]:
```suggestion
trap
```

File: xml/hwbp_registers.xml [hunk 5]
```diff
@@ -242,7 +251,7 @@
             1: Perform a match on the data value loaded or stored, or the
             instruction executed.
         </field>
-        <field name="timing" bits="18" access="R/W" reset="0">
+        <field name="timing" bits="18" access="WARL" reset="0">
```
Comment (ernie-sifive) [inline xml/hwbp_registers.xml]: If you're going to change this to WARL, why not the rest of the fields in this register?  They can only take on certain legal values too.

Comment (marceg) [reply xml/hwbp_registers.xml]: Perhaps maskmax is the only field not WARL, because it's not writable.

Comment (timsifive) [reply xml/hwbp_registers.xml]: Good point. I changed almost all of them.

**Pull request merged**